### DISCUSSION TITLE
repatching OTA

### DISCUSSION THREAD

**Discussion by c0mpile** [chenxiaolong/avbroot#313] [Q&A] [answered]
Am I correct in assuming that using the same process in the repair mode section of the readme, I can root and unroot at will just by flashing the rooted and unrooted OTAs respectively? I want to be able to use rooted /data backup and restore with SwiftBackup but disable root when I'm not actively doing the backups.

**Comment by chenxiaolong** [comment, marked as answer]
Yep! You can freely switch between rooted and `--rootless` patched OTAs any time.

**Reply by c0mpile** [reply]
You're my new favorite person.

**Reply by e-t-l** [reply]
this is actually a really smart idea. Very nice balance between the convenience vs security tradeoff of root vs non-root. I've also seen people flash one rooted OS image and one unrooted OS image into the A/B system partitions, effectively allowing you to dual-boot into either. Not sure the precise steps to do this or how they switch from one to the other, but it's a neat idea.

**Reply by chenxiaolong** [reply]
> I've also seen people flash one rooted OS image and one unrooted OS image into the A/B system partitions, effectively allowing you to dual-boot into either. Not sure the precise steps to do this or how they switch from one to the other, but it's a neat idea.

This is unfortunately not possible anymore on newer devices that use Android's Virtual A/B setup. VAB devices no longer reserve double the space to hold both the A and B slots. When installing an OTA update, Android temporarily creates a "virtual slot" uses whatever remaining space is available in the `super` partition and spilling over to the free space on the `/data` partition.

After the OTA update is complete and the device successfully boots into the temporary slot, Android will take 5-10 minutes and in the background, merge the new slot's data back into the `super` partition and then free up the space, leaving the device with 1 actual slot again.

(and if the OTA fails or is unbootable, Android just keeps the original slot and discards the temporary slot)
